File: CS 471/program2.pl
#Joseph Camacho-Terrazas
# 09/6/2020
# Input: None
# Output: Print statements from tests.
# Pre-condition: None
# Post-condition: Print statements from tests to see if Evaluate is called in comparison statements.

#!/usr/bin/perl

use warnings;

sub Evaluate {
	printf "Condition has been evaluated \n";
	return 1;
}

$a = 1;

#Testing "AND" logical operators
printf "===Testing AND===\n";
#Test F and T
if($a == 0 && Evaluate() == 1){
	printf "True\n";
}else{
	printf "False\n"
}

printf "\n";

#Test T and F
if($a == 1 && Evaluate() == 0){
	printf "True\n";
}else{
	printf "False\n";
}

printf "\n";

#Test T and T
if($a == 0 && Evaluate() == 0){
	printf "True\n";
}else{
	printf "False\n";
}

printf "\n";

#Test F and F
if($a == 1 && Evaluate() == 1){
	printf "True\n";
}else{
	printf "False\n";
}

printf "\n";

#Testing "OR" logical operators
printf "===Testing OR===\n";
#Test F and T
if($a == 0 || Evaluate() == 1){
	printf "True\n";
}else{
	printf "False\n";
}

printf "\n";

#Test T and F
if($a == 1 || Evaluate() == 0){
	printf "True\n";
}else{
	printf "False\n";
}

printf "\n";

#Test T and T
if($a == 1 || Evaluate() == 1){
	printf "True\n";
}else{
	printf "False\n";
}

printf "\n";

#Test F and F
if($a == 0 || Evaluate() == 0){
	printf "True\n";
}else{
	printf "False\n";
}

File: CS 471/program2.php
<?php  
//Joseph Camacho-Terrazas
//9/6/2020
//Input:
//Output:
//Preconditions: None
//Postconditions: 

//Create Evaluate function to test when the second part of the statements are executed.
function Evaluate(){
	echo "Condition has been evaluated" .PHP_EOL;
	return True;
	
}

//Testing "AND" operators
echo "===Testing AND===" .PHP_EOL;
//Test F and T
if (False && Evaluate()) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;

//Test T and F
if (True && Evaluate() == False) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;

//Test T and T
if (True && Evaluate()) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;
	
//Test F and F
if (False && Evaluate() == False) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;

//Testing "OR" operators
echo "===Testing OR===" .PHP_EOL;
//Test F and T
if (False || Evaluate()) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;
	
//Test T and F
if (True || Evaluate() == False) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;

//Test T and T
if (True || Evaluate()) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

echo "" .PHP_EOL;

//Test F and F
if (False || Evaluate() == False) {
	echo "True" .PHP_EOL;
} else {
	echo "False" .PHP_EOL;
}

?> 